converter/src/main/java/org/dswarm/converter/schema/SolrSchemaParser.java

```diff
@@ -0,0 +1,372 @@
+
+		final NodeList fields = optionalFields.get();
+		final List<Attribute> attributes = determineAndCreateAttributes(fields, schemaBaseURI);
+
+		if (attributes.isEmpty()) {
+
+			LOG.error("could not extract any attribute from the Solr schema at '" + solrSchemaFilePath + "'");
+
+			return Optional.empty();
+		}
+
+		// attribute paths
+		final List<AttributePath> attributePaths = createAttributePaths(attributes);
+
+		if (attributePaths.isEmpty()) {
+
+			LOG.error("couldn't create any attribute path from the extracted attributes from the Solr schem at '" + solrSchemaFilePath + "'");
+
+			return Optional.empty();
+		}
+
+		// schema attribute paths
+		for (final AttributePath attributePath : attributePaths) {
+
+			final SchemaAttributePathInstance schemaAttributePathInstance = createSchemaAttributePathInstance(attributePath);
+
+			schema.addAttributePath(schemaAttributePathInstance);
+		}
+
+		// record class
+		final String recordClassURI = schemaBaseURI + DEFAULT_RECORD_CLASS_LOCAL_NAME;
+
+		SchemaUtils.addRecordClass(schema, recordClassURI, classServiceProvider);
+
+		final Optional<ProxySchema> optionalProxySchema = Optional.ofNullable(schemaServiceProvider.get().createObjectTransactional(schema));
+
+		if (!optionalProxySchema.isPresent()) {
+
+			return Optional.empty();
+		}
+
+		return Optional.ofNullable(optionalProxySchema.get().getObject());
+	}
+
+	private Optional<Document> readXML(final String solrSchemaFilePath) {
+
+		final DocumentBuilder documentBuilder;
+
+		try {
+
+			documentBuilder = documentBuilderFactory.newDocumentBuilder();
+		} catch (final ParserConfigurationException e) {
+
+			LOG.error("couldn't create document builder for parsing the Solr schema file", e);
+
+			return Optional.empty();
+		}
+
+		final InputStream inputStream = ClassLoader.getSystemResourceAsStream(solrSchemaFilePath);
+
+		final Document document;
+
+		try {
+			document = documentBuilder.parse(inputStream);
+
+			return Optional.ofNullable(document);
+		} catch (final SAXException e) {
+
+			LOG.error("couldn't parse the Solr schema file at '" + solrSchemaFilePath + "'", e);
+
+			return Optional.empty();
+		} catch (final IOException e) {
+
+			LOG.error("couldn't read the Solr schema fila at '" + solrSchemaFilePath + "'", e);
+
+			return Optional.empty();
+		}
+	}
+
+	private Optional<NodeList> getFields(final Document document, final String solrSchemaFilePath) {
+
+		final XPath xpath = xPathfactory.newXPath();
+
+		final XPathExpression expr;
+
+		try {
+
+			expr = xpath.compile(FIELDS_XPATH_EXPRESSION);
+		} catch (final XPathExpressionException e) {
+
+			LOG.error("could not compile xpath", e);
+
+			return Optional.empty();
+		}
+
+		final NodeList result;
+
+		try {
+
+			result = (NodeList) expr.evaluate(document, XPathConstants.NODESET);
+		} catch (final XPathExpressionException e) {
+
+			LOG.error("could not execute xpath query against document (from '" + solrSchemaFilePath + "')", e);
```

Comment:
`LOG.error("could not execute xpath query against document (from '{}')", solrSchemaFilePath, e);`